client/fingerprint/host.go

```diff
@@ -28,13 +29,19 @@ func (f *HostFingerprint) Fingerprint(req *FingerprintRequest, resp *Fingerprint
 		f.logger.Warn("error retrieving host information", "error", err)
 		return err
 	}
-
+	print(hostInfo.PlatformVersion)
+
+	if runtime.GOOS == "windows" {
+		platformVersion := strings.Split(hostInfo.PlatformVersion, "Build")
+		buildNumber := strings.TrimSpace(platformVersion[1])
+		resp.AddAttribute("os.build", buildNumber)
```

Comment:
We don't control the input here, so we can't rely 100% on the string "Build" not being changed by upstream. If it changes and we drop the `os.build` attribute, that'll be no big deal but with the code as written here we'd actually panic the client. So we need to check the length of `platformVersion` before accessing `[1]` and trying to assign the attribute.

Also, do we actually want to throw away the leading `10.0` value? Isn't that still a valid value for the `os.version` attribute?

Reply:
agreed !!